src/prototype_room_basebuilder.js

```diff
@@ -67,7 +67,7 @@ Room.prototype.destroyStructure = function(structure) {
     structuresMin = 1;
   }
 
-  if (structures.length > structuresMin) {
+  if (structures.length > structuresMin && Room.structureIsEmpty(structure)) {
```

Comment:
Overall I like the changes and I think it is fine to merge. Just this line I'm pretty sure this will break the cases where the room layout is changed and structures need to be moved to somewhere else.